src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,57 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+
+/**
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime localDateTime;
+    protected String date;
+
+    private DateTimeFormatter fmt = new DateTimeFormatterBuilder()
+            .appendPattern("[d-MM-yyyy HHmm][yyyy-MM-d HHmm]")
+            .optionalStart()
+            .appendPattern(" HHmm")
+            .optionalEnd()
+            .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+            .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+            .toFormatter();
+
+    /**
+     * Creates a new instance of a Deadline.
+     *
+     * @param description Description of the task with deadline.
+     * @param by Date of the task's deadline.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.date = by;
```

Comment:
I think you can consider using dateBy instead of by so it's clearer what the variable is referring to.